contrib/fedora/rpm/firewalld.spec

```diff
@@ -0,0 +1,412 @@
+Requires: NetworkManager-libnm
+Requires: dbus-x11
+Recommends: polkit
+
+%description -n firewall-config
+The firewall configuration application provides an configuration interface for
+firewalld.
+
+%if 0%{?rhel} && 0%{?rhel} < 10
+%pretrans -p <lua>
+-- HACK: Old rpm versions had an untracked (%ghost) symlink for
+-- /etc/firewalld/firewalld.conf. RPM won't handle replacing the symlink due to
+-- "%config(noreplace)". As such, we remove the symlink here before attempting
+-- to install the new version which is a real file. Only replace the symlink if
+-- the target matches one of the previous package's expected targets.
+--
+-- Unfortunately this must be done in pretrans in order to occur before RPM
+-- makes decisions about file replacement.
+--
+local old_package_symlinks = {"firewalld-standard.conf", "firewalld-server.conf",
+                              "firewalld-workstation.conf"}
+
+local symlink_target = posix.readlink("%{_sysconfdir}/firewalld/firewalld.conf")
+for k,v in ipairs(old_package_symlinks) do
+  if symlink_target == v then
+    posix.unlink("%{_sysconfdir}/firewalld/firewalld.conf")
+    break
+  end
+end
+%endif
+
+%prep
+%autosetup -p1
+
+%build
+%configure --enable-sysconfig --enable-rpmmacros PYTHON="%{__python3} %{py3_shbang_opts}"
+make %{?_smp_mflags}
+
+%install
+make install DESTDIR=%{buildroot}
+
+desktop-file-install --delete-original \
+  --dir %{buildroot}%{_sysconfdir}/xdg/autostart \
+  %{buildroot}%{_sysconfdir}/xdg/autostart/firewall-applet.desktop
+desktop-file-install --delete-original \
+  --dir %{buildroot}%{_datadir}/applications \
+  %{buildroot}%{_datadir}/applications/firewall-config.desktop
+
+install -d -m 755 %{buildroot}%{_prefix}/lib/firewalld/zones/
+
+%if 0%{?fedora}
+install -c -m 644 %{SOURCE1} %{buildroot}%{_prefix}/lib/firewalld/zones/FedoraServer.xml
+install -c -m 644 %{SOURCE2} %{buildroot}%{_prefix}/lib/firewalld/zones/FedoraWorkstation.xml
+%endif
+
+%if %{with fedora_variant}
+install -m 644 -D %{SOURCE3} %{buildroot}%{_datadir}/polkit-1/rules.d/org.fedoraproject.FirewallD1.desktop.rules.choice
+%endif
+
+%if %{with fedora_variant}
+# standard firewalld.conf
+mv %{buildroot}%{_sysconfdir}/firewalld/firewalld.conf \
+    %{buildroot}%{_sysconfdir}/firewalld/firewalld-standard.conf
+
+# server firewalld.conf
+cp -a %{buildroot}%{_sysconfdir}/firewalld/firewalld-standard.conf \
+    %{buildroot}%{_sysconfdir}/firewalld/firewalld-server.conf
+sed -i 's|^DefaultZone=.*|DefaultZone=FedoraServer|g' \
+    %{buildroot}%{_sysconfdir}/firewalld/firewalld-server.conf
+
+# workstation firewalld.conf
+cp -a %{buildroot}%{_sysconfdir}/firewalld/firewalld-standard.conf \
+    %{buildroot}%{_sysconfdir}/firewalld/firewalld-workstation.conf
+sed -i 's|^DefaultZone=.*|DefaultZone=FedoraWorkstation|g' \
+    %{buildroot}%{_sysconfdir}/firewalld/firewalld-workstation.conf
+
+rm -f %{buildroot}%{_datadir}/polkit-1/actions/org.fedoraproject.FirewallD1.policy
+%endif
+
+%find_lang %{name} --all-name
```

Comment:
I have no idea what this does. Can you explain it?

Reply:
I don't know. You find it both in [Fedora]( It comes from https://src.fedoraproject.org/rpms/firewalld/c/4d63fa14ad0ef462713fd3b8fee80a68adca78e0) and [RHEL](https://gitlab.com/redhat/centos-stream/rpms/firewalld/-/blob/dcf3c6db0375bd2f041674259423fe92d7a0dc74/firewalld.spec#L131).

Maybe: https://fedoraproject.org/wiki/PackagingDrafts/find_langt